br: add doc for latest backup protection (#19639)
pingcap/docs: tidb-cloud/backup-and-restore.md

        > **Note**
        >

>         > - All auto-backups, except the latest one, will be deleted if their lifetime exceeds the retention period. The latest auto-backup will not be deleted unless you delete it explicitly. This ensures that a cluster still has data to restore after it is accidentally deleted.

Reviewer: ```suggestion
        > - All auto-backups, except the latest one, will be deleted if their lifetime exceeds the retention period. The latest auto-backup will not be deleted unless you delete it manually. This ensures that you can restore cluster data if accidental deletion occurs.
```

>         > - All auto-backups, except the latest one, will be deleted if their lifetime exceeds the retention period. The latest auto-backup will not be deleted unless you delete it explicitly. This ensures that the cluster is not accidentally deleted without the ability to restore data.

Reviewer 2: ```suggestion
        > - All auto-backups, except the latest one, will be deleted if their lifetime exceeds the retention period. The latest auto-backup will be retained unless you delete it explicitly. This ensures that the cluster is not accidentally deleted without the ability to restore data.
```

>         > - All auto-backups, except the latest one, will be deleted if their lifetime exceeds the retention period. The latest auto-backup will not be deleted unless you delete it explicitly. This ensures that the cluster is not accidentally deleted without the ability to restore data.

Reviewer 2: ```suggestion
        > - All auto-backups, except the latest one, will be deleted if their lifetime exceeds the retention period. The latest auto-backup will not be deleted unless you delete it explicitly. This ensures that a cluster still has data to restore after it is accidentally deleted.
```

>         > - After you delete a cluster, auto-backups with a lifetime within the retention period will be moved to the Recycle Bin. 

Reviewer: ```suggestion
        > - After you delete a cluster, auto-backups with a lifetime within the retention period will be moved to the recycle bin.
```
Keep the capitalization consistent with:
https://github.com/pingcap/docs/blob/a0100a3605f5e062367e1d8d3b3017c412986429/tidb-cloud/backup-and-restore.md?plain=1#L9

>         > - After you delete a cluster, auto-backups with a lifetime within the retention period will be moved to the Recycle Bin. Auto-backups, except the latest one, will be deleted if their lifetime exceeds the retention period. The latest auto-backup will not be deleted unless you delete it explicitly. This prevents accidental cluster deletion with no data to restore.

Reviewer 2: ```suggestion
        > - After you delete a cluster, auto-backups with a lifetime within the retention period will be moved to the Recycle Bin. 
```

>         > - Auto-backups except the latest auto-backup will be deleted if the lifetime exceed the retention period. For the latest auto-backup, it will not be deleted until you delete it explicitly. This will prevent the cluster is deleted mistakenly and no data to restore.
>         > - After you delete a cluster, auto-backup which lifetime is not exceed the retention period will be retained in Recycle Bin. Auto-backups except the latest auto-backup will be deleted if the lifetime exceed the retention period. For the latest auto-backup, it will not be deleted until you delete it explicitly. This will prevent the cluster is deleted mistakenly and no data to restore.
>         > - After you delete a cluster, the existing manual backups will be retained until you manually delete them, or your account is closed.

Reviewer 2: ```suggestion
        > - All auto-backups, except the latest one, will be deleted if their lifetime exceeds the retention period. The latest auto-backup will not be deleted unless you delete it explicitly. This ensures that the cluster is not accidentally deleted without the ability to restore data.
        > - After you delete a cluster, auto-backups with a lifetime within the retention period will be moved to the Recycle Bin. Auto-backups, except the latest one, will be deleted if their lifetime exceeds the retention period. The latest auto-backup will not be deleted unless you delete it explicitly. This prevents accidental cluster deletion with no data to restore.
        > - After you delete a cluster, existing manual backups will be retained until manually deleted or your account is closed.
```
